plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/ServicesAndRoutingPage.java

```diff
@@ -92,11 +103,40 @@ protected void doCreateControls(Composite parent, DataBindingContext dbc) {
 		btnAddRoute.setText("Add Route");
 		btnAddRoute.setToolTipText("Adding a route to the service will make the image accessible\noutside of the OpenShift cluster on all the available service ports. \nYou can target a specific port by editing the route later.");
 		GridDataFactory.fillDefaults()
-			.align(SWT.FILL, SWT.FILL).grab(false, false).applyTo(btnAddRoute);
+			.align(SWT.FILL, SWT.FILL).grab(false, false).span(2, 1).applyTo(btnAddRoute);
 		ValueBindingBuilder.bind(WidgetProperties.selection().observe(btnAddRoute))
 			.to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ADD_ROUTE)
 			.observe(model))
 			.in(dbc);
+		
+		Label labelRouteHostname = new Label(routingContainer, SWT.NONE);
+		labelRouteHostname.setText("Hostname:");
+	    GridDataFactory.fillDefaults()
+          .align(SWT.FILL, SWT.FILL)
+          .applyTo(labelRouteHostname);
+
+		Text textRouteHostname = new Text(routingContainer, SWT.BORDER);
+        GridDataFactory.fillDefaults()
+        .align(SWT.FILL, SWT.CENTER)
+        .grab(true, false)
+        .applyTo(textRouteHostname);
+        ValueBindingBuilder.bind(WidgetProperties.enabled().observe(textRouteHostname))
+        .to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ADD_ROUTE)
+        .observe(model))
+        .in(dbc);
+        ValueBindingBuilder.bind(WidgetProperties.text(SWT.Modify).observe(textRouteHostname))
+        .converting(new TrimmingStringConverter())
+        .validatingAfterConvert(new IValidator() {
+            
+            @Override
+            public IStatus validate(Object value) {
+                boolean valid = DomainValidator.getInstance(true).isValid((String) value);               
+                return valid?ValidationStatus.ok():ValidationStatus.error(NLS.bind(OpenShiftUIMessages.InvalidHostNameErrorMessage, value));
+            }
+        })
+        .to(BeanProperties.value(IServiceAndRoutingPageModel.PROPERTY_ROUTE_HOSTNAME)
+        .observe(model))
+        .in(dbc);
```

Comment:
Validation wont only happen if "Add Route" is checked. It is always executed. 
![image](https://cloud.githubusercontent.com/assets/25126/19072532/01521e9a-8a36-11e6-8c36-d47cba664387.png)
Use a MultiValidator that tracks the checkbox and the textfield for this sake.